packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_controller.dart

```diff
@@ -527,40 +529,68 @@ class DeepLinksController extends DisposableController
           domains: domains,
         );
         iosDomainErrors = iosResult.domainErrors;
+        iosDomainPaths = iosResult.paths;
       }
     } catch (_) {
       // TODO(hangyujin): Add more error handling for cases like RPC error and invalid json.
       pagePhase.value = PagePhase.validationErrorPage;
-      return linkdatas;
+      return rawLinkdatas;
     }
 
-    return linkdatas.map((linkdata) {
+    final validatedLinkDatas = <LinkData>[];
+
+    for (final linkdata in rawLinkdatas) {
       final errors = <DomainError>[
         if (linkdata.os.contains(PlatformOS.android))
           ...(androidDomainErrors[linkdata.domain] ?? []),
         if (linkdata.os.contains(PlatformOS.ios))
           ...(iosDomainErrors[linkdata.domain] ?? []),
       ];
-      if (errors.isNotEmpty) {
-        return LinkData(
-          domain: linkdata.domain,
-          domainErrors: errors,
-          path: linkdata.path,
-          pathErrors: linkdata.pathErrors,
-          os: linkdata.os,
-          scheme: linkdata.scheme,
-          associatedDomains: linkdata.associatedDomains,
-          associatedPath: linkdata.associatedPath,
-          hasAndroidAssetLinksFile: !(androidDomainErrors[linkdata.domain]
-                  ?.contains(AndroidDomainError.existence) ??
-              false),
-          hasIosAasaFile: !(iosDomainErrors[linkdata.domain]
-                  ?.contains(IosDomainError.existence) ??
-              false),
+      final hasAndroidAssetLinksFile = !(androidDomainErrors[linkdata.domain]
+              ?.contains(AndroidDomainError.existence) ??
+          false);
+      final hasIosAasaFile = !(iosDomainErrors[linkdata.domain]
+              ?.contains(IosDomainError.existence) ??
+          false);
```

Comment:
is defaulting to false what we want here? The way I am understanding this is that if we have no domain errors, we assume the asset links or aasa files do not exist.

Reply:
If we have no domain errors, hasAndroidAssetLinksFile =! ( null ?? false ) = true .

if the asset links or aasa files do not exist, the domain errors should return AndroidDomainError.existence or  IosDomainError.existence.

If we have no domain errors, we assume the asset links or aasa files do exist.